lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -1794,12 +1826,32 @@ GLStatus TFGraphLowering::visitGraphOperationInst(GraphOperationInst *inst) {
       // Done with normal attributes.
       break;
     case SILTensorOpInfo::OperandClass::DType: {
-      // This integer value is a dtype.
-      auto val = attrValue.getIntegerValue();
-      TF_SetAttrType(op, name.data(), (TF_DataType)val.getLimitedValue());
+      auto swiftType = attrValue.getMetatypeValue();
+      auto tfType = convertSwiftTypeToTF(swiftType);
+      TF_SetAttrType(op, name.data(), (TF_DataType)tfType);
+      break;
+    }
+    case SILTensorOpInfo::OperandClass::Tensor: {
+      // Tensor can support two cases: an array case (not yet implemented), and
+      // a scalar case.
+      TF_DataType dtype;
+      SmallVector<APInt, 4> elements;
+      SmallVector<int64_t, 4> shape;
+      if (attrValue.getKind() != SymbolicValue::Integer) {
+        llvm_unreachable("FIXME: Tensor-typed attr is not yet handled");
+      }
+      // The scalar case is very simple, the shape of a scalar is 0d, and the
+      // data type is int64.
+      dtype = TF_INT64;
```

Comment:
`dtype` is created above but it's not used until this line here. Can it be declared here instead?

Reply:
This function will be extended in a later PR to support the "tensor-type" case, where I believe we will need to define `dtype` earlier (as in the counterpart code in `TFGraphLowering::visitTFOpInst()`). 

If that does not happen, I'll refactor the code here. Thanks!